server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/Main.scala

```diff
@@ -21,19 +21,20 @@ object Main extends IOApp {
 
     val app: Stream[IO, Unit] = for {
       blockingEc <- Stream.resource[IO, ExecutionContext](ExecutionContexts.cachedThreadPool)
+      blocker = Blocker.liftExecutionContext(blockingEc)
       appConfig <- Stream.fromEither[IO](Config.appConfig)
       storageLinksCache <- cachedResource[RelativePath, StorageLink](
         appConfig.pathToStorageLinksJson,
-        blockingEc,
+        blocker,
         storageLink => List(storageLink.localBaseDirectory.path -> storageLink, storageLink.localSafeModeBaseDirectory.path -> storageLink)
       )
       metadataCache <- cachedResource[RelativePath, AdaptedGcsMetadataCache](
         appConfig.pathToGcsMetadataJson,
-        blockingEc,
+        blocker,
         metadata => List(metadata.localPath -> metadata)
       )
-      streams <- initStreams(appConfig, blockingEc, storageLinksCache, metadataCache)
-      _ <- Stream.emits(streams).covary[IO].parJoin(streams.length)
+      streams <- initStreams(appConfig, blocker, storageLinksCache, metadataCache)
+      _ <- Stream.emits(streams).covary[IO].parJoin(6)
```

Comment:
this seems like a magic number

Reply:
ah, this is not intended..it should be `streams.length`